### PR TITLE
Acceptance_ratio and average_consumers: Fixed typos and included tests

### PR DESCRIPTION
**Pull request by MantautasRimkus**
Sorry that this one has two commits (and two tests) instead of one. I forgot that if I push a new commit, pull request automatically updates. 
This commit contains:
1. Corrected acceptance_ratio (I saw your updates, they look great. Is just I found one mistake and fixed it. Also, I included a test for this consumer.
2. Corrected average_cosinus (during the implementation of test, I found one small typo). Also, this commit comes with tests for this consumer.

### CODE REVIEW

File: include/sampleflow/consumers/acceptance_ratio.h [hunk 1]
```diff
@@ -203,6 +203,7 @@ namespace SampleFlow
               // the new sample.
               ++n_accepted_samples;
               previous_sample = sample;
+              ++n_samples;
```
Comment (bangerth) [inline include/sampleflow/consumers/acceptance_ratio.h]: Oops -- you're entirely right!

File: include/sampleflow/consumers/average_cosinus.h [hunk 2]
```diff
@@ -168,7 +168,7 @@ namespace SampleFlow
                 }
               update = update/(std::sqrt(norm1*norm2));
               update -= current_avg_cosine[i];
-              update /= static_cast<double>(n_samples-i);
+              update /= static_cast<double>(n_samples-i-1);
```
Comment (bangerth) [inline include/sampleflow/consumers/average_cosinus.h]: I'll trust you on this one :-)

Comment (MantautasRimkus) [reply include/sampleflow/consumers/average_cosinus.h]: "i" starts at 0, while we minimum degrees of freedom we lose is 1. With big n, doesn't make almost no difference.

**Comment by MantautasRimkus** [comment]
The requested change is done.

**Pull request merged**